### ISSUE TITLE
a11y: Agencies filter search box findings

### ISSUE THREAD

**Issue by iamjolly** [18F/all_sorns#155]
## Description:
The proposed agency filter search bar to narrow the list of agencies to filter by has three aspects which will need some attention to make it fully accessible. Those are:
1. Search/filter field is unlabeled. 
    - Add a `label` for the search `input`, preferably a visual label so sighted and screen reader users alike will understand the intended functionality.
2. No indication is provided when the list changes based on user input. 
    - Add messaging when the list changes based on what is typed into the search box. This is another instance where a visual indication may be handy for some users (see screenshot of proposed approach), but it could be presented as a visually-hidden area for screen reader users only.
3. When a user hits <enter>, the page reloads and their filter is not preserved because the form submits. 
    - Since the updates to the results list are in real-time as a user types, the default form action (submit) should be suppressed if a user presses <enter/return>. Adding [`event.preventDefault()`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) to the element/function may prevent this issue.

### Screenshot of current unlabeled field
<img width="1228" alt="agencies-filter-search" src="https://user-images.githubusercontent.com/1093423/105918562-5a334a80-5ff1-11eb-84dc-62dc9699b58d.png">

### Proposed approach to adding a live region for filter search results:
<img width="1282" alt="agencies-filter-live-status" src="https://user-images.githubusercontent.com/1093423/105918496-3f60d600-5ff1-11eb-8126-7e3859717400.png">

## Recommended code for live region for results:
Note: The `div` for the live region, with `aria` attributes must be loaded in the DOM with the page, not inserted into the DOM by javascript. When the contents change within it, it will then announce the contents with each update.
```
<div class="usa-prose" aria-live="polite" aria-atomic="true">SEARCH_TERM has NUMBER_OF_RESULTS agency matches</div>
```

## WCAG SC
- [Success Criterion 3.3.2 Labels or Instructions (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions): Labels or instructions are provided when content requires user input.
- [Success Criterion 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value): For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.
- [Success Criterion 4.1.3 Status Messages (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/status-messages): In content implemented using markup languages, status messages can be programmatically determined through role or properties such that they can be presented to the user by assistive technologies without receiving focus.
